arungeorge101/myRobinhoodData
# Refactored to OOP, Upgraded to Py3, Added Pylint

Hello! I was interested in contributing to this repo after seeing #3 as an open issue. It spiraled into this, with the same end behavior but refactored.

## Overview of Changes
- Refactored logic into a class `RobinhoodAPIWrapper` with a few helper functions for exporting to HTML or XLS files
- Fixes #3 - Upgraded the code to be Python3 compatible
- Added Pylint for code standardization

## How to Test
Checkout my branch and run `python3 src/main.py` after setting up `token.txt` based on the instructions in the new README. You should have 6 files appear in the `results` directory - 3 JSON files and 3 HTML files - with your Robinhood watchlist, orders history, and options history.

There are no formal Python unit tests set up.

## Suggestions for additional issues
After performing this rewrite, here are my suggestions for future tasks to look into once merged:
- Feature: Create a helper script that automatically extracts a user's Robinhood API bearer token from their browser into token.txt without needing to perform the process of opening developer tools entirely.
- Feature: Re-implement for the Summary sections of each dataset: watchlist, orders, options.
- Meta: Publish RobinhoodAPIWrapper (renamed to something more appropriate or catchy) as a Python package to `pypi.org` for easier install.
- Feature: Add badge to README to indicate `Pylint` GitHub Action workflow passing status (I can paste my code snippet for doing so once issue is created).
